test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/RequestCompressionTest.java

```diff
@@ -0,0 +1,150 @@
+        PutOperationWithRequestCompressionRequest request =
+            PutOperationWithRequestCompressionRequest.builder()
+                                                     .body(SdkBytes.fromByteArray(uncompressedData))
+                                                     .overrideConfiguration(o -> o.requestCompressionConfiguration(
+                                                         c -> c.minimumCompressionThresholdInBytes(1)))
+                                                     .build();
+        syncClient.putOperationWithRequestCompression(request);
+
+        SdkHttpFullRequest loggedRequest = (SdkHttpFullRequest) mockHttpClient.getLastRequest();
+        InputStream loggedStream = loggedRequest.contentStreamProvider().get().newStream();
+        byte[] loggedBody = SdkBytes.fromInputStream(loggedStream).asByteArray();
+        int loggedSize = Integer.valueOf(loggedRequest.firstMatchingHeader("Content-Length").get());
+        byte[] compressedData = compressor.compress(uncompressedData);
+
+        assertThat(loggedBody).isEqualTo(compressedData);
+        assertThat(loggedSize).isEqualTo(compressedData.length);
+        assertThat(loggedRequest.firstMatchingHeader("Content-encoding").get()).isEqualTo("gzip");
+    }
+
+    @Test
+    public void async_nonStreaming_compression_compressesCorrectly() throws InterruptedException {
+        mockAsyncHttpClient.stubNextResponse(mockResponse(), Duration.ofMillis(500));
+
+        byte[] uncompressedData = COMPRESSABLE_BODY.getBytes(StandardCharsets.UTF_8);
+        PutOperationWithRequestCompressionRequest request =
+            PutOperationWithRequestCompressionRequest.builder()
+                                                     .body(SdkBytes.fromByteArray(uncompressedData))
+                                                     .overrideConfiguration(o -> o.requestCompressionConfiguration(
+                                                         c -> c.minimumCompressionThresholdInBytes(1)))
+                                                     .build();
+
+        asyncClient.putOperationWithRequestCompression(request);
+
+        SdkHttpFullRequest loggedRequest = (SdkHttpFullRequest) mockAsyncHttpClient.getLastRequest();
+        InputStream loggedStream = loggedRequest.contentStreamProvider().get().newStream();
+        byte[] loggedBody = SdkBytes.fromInputStream(loggedStream).asByteArray();
+        int loggedSize = Integer.valueOf(loggedRequest.firstMatchingHeader("Content-Length").get());
+        byte[] compressedData = compressor.compress(uncompressedData);
+
+        assertThat(loggedBody).isEqualTo(compressedData);
+        assertThat(loggedSize).isEqualTo(compressedData.length);
+        assertThat(loggedRequest.firstMatchingHeader("Content-encoding").get()).isEqualTo("gzip");
+    }
+
+    @Test
+    public void sync_streaming_compression_compressesCorrectly() throws InterruptedException {
```

Comment:
Could you please add test cases for error retries  ? something like stubForFailureThenSuccess()

Reply:
added tests for retries, stubbed error responses